shopinvader_api_address/models/res_partner.py

```diff
@@ -0,0 +1,128 @@
+            "phone": data.phone or "",
+            "email": data.email or "",
+            "state_id": state_id.id,
+            "country_id": country_id.id,
+            "parent_id": self.env.context.get("authenticated_partner_id"),
+            "type": data.type,
+        }
+        return vals
+
+    @api.model
+    def _create_shopinvader_address(self, data: AddressInput) -> "ResPartner":
+        vals = self._create_vals_shopinvader_address(data)
+        return self.env["res.partner"].create(vals)
+
+    @api.model
+    def _get_shopinvader_address(self, rec_id: int) -> "ResPartner":
+        address = self.env["res.partner"].search([("id", "=", rec_id)], limit=1)
+        if not address:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail=_("No address found")
+            )
+        return address
+
+    @api.model
+    def _build_shopinvader_search_address_domain(self, query: AddressSearch) -> list:
+        domain = []
+
+        if query.name is not None:
+            domain.append(("name", "ilike", query.name))
+        if query.street is not None:
+            domain.append(("street", "ilike", query.name))
+        if query.street2 is not None:
+            domain.append(("street2", "ilike", query.name))
+        if query.zip is not None:
+            domain.append(("zip", "ilike", query.name))
+        if query.city is not None:
+            domain.append(("city", "ilike", query.name))
+        if query.phone is not None:
+            domain.append(("phone", "ilike", query.name))
+        if query.email is not None:
+            domain.append(("email", "ilike", query.name))
+        if query.type is not None:
+            domain.append(("type", "=", query.type))
+
+        # search on state/country is performed
+        # on name or code of state/country
+        if query.state is not None:
+            domain.join(
+                [
+                    "|",
+                    ("state_id.name", "ilike", query.state),
+                    ("state_id.code", "ilike", query.state),
+                ]
+            )
+        if query.country is not None:
+            domain.join(
+                [
+                    "|",
+                    ("country.name", "ilike", query.country),
+                    ("country.code", "ilike", query.country),
+                ]
+            )
+
+        return domain
+
+    @api.model
+    def _search_shopinvader_address(
+        self, query: AddressSearch, limit, offset
+    ) -> "ResPartner":
+        domain = self._build_shopinvader_search_address_domain(query)
+        return self.env["res.partner"].search(domain, limit=limit, offset=offset)
+
+    @api.model
+    def _delete_shopinvader_address(self, rec_id: int):
+        address = self.env["res.partner"].search([("id", "=", rec_id)], limit=1)
+        if not address:
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND, detail=_("No address found")
```

Comment:
use odoo esception insteda of httpexception?

Reply:
For sure!

Reply:
Standard odoo exception are translated into the right http exception by error handlers on the fastapi app... see https://github.com/OCA/rest-framework/blob/16.0/fastapi/error_handlers.py